pkg/spanconfig/spanconfigkvsubscriber/kvsubscriber.go

```diff
@@ -248,15 +249,25 @@ func (s *KVSubscriber) handlePartialUpdate(
 	handlers := s.mu.handlers
 	s.mu.Unlock()
 
-	for _, h := range handlers {
+	for i := range handlers {
 		for _, ev := range events {
-			// TODO(arul): In the future, once we start reacting to system span
-			// configurations, we'll want to invoke handlers with the correct span
-			// here as well.
 			target := ev.(*bufferEvent).Update.Target
-			if target.IsSpanTarget() {
-				h.invoke(ctx, target.GetSpan())
+			var sp roachpb.Span
+			switch {
+			case target.IsSpanTarget():
+				sp = target.GetSpan()
+			case target.IsSystemTarget():
+				var err error
+				sp, err = target.GetSystemTarget().KeyspaceTargeted()
+				if err != nil {
+					// Swallow the error here, even though we never expect this to happen.
```

Comment:
This is still a bit odd, and tightly coupled to internals of KeyspaceTargeted. I think you want a check at this level that the SystemTargetType isn't the odd one you're not expecting and fatal-ing if it is. With that, KeyspaceTargeted could `panic("unimplemented")` or `panic("not applicable")`. I should've been clearer earlier, I was just thrown off by the "read only target" message in the panic.